src/main/kotlin/com/hibob/academy/types/StoreService.kt

```diff
@@ -0,0 +1,80 @@
+package com.hibob.academy.types
+
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+import Cart
+import Payment
+import Check
+import Statuses
+
+const val ZERO_TOTAL_COSTS_ON_FAILED_BUY = 0.0
+const val VISA_LEGIT_NUM_LENGTH = 10
+
+fun fail(message: String) : Nothing?{
+    throw IllegalStateException(message)
+}
+
+@Component
+class StoreService {
+
+    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
+        val result = mutableMapOf<String, Check>()
+
+        for (c in cart) {
+            val check = checkout(c, payment)
+
+            result[c.clientId] = check
+        }
+        return result
+    }
+
+    private fun isCustomTrue(custom: Any): Boolean {
+
+        return when(custom) {
+            is Boolean -> custom
+            else -> false
+        }
+    }
+
+    private fun checkout(cart: Cart, payment: Payment): Check {
+        var totalPricesOfLegitProducts = 0.0
+
+        cart.products.forEach { product ->
+            if (isCustomTrue(product.custom)) {
+                totalPricesOfLegitProducts += product.price
+            }
+        }
+
+        val status: Statuses = when(payment) {
+            is Payment.Cash -> {
+                fail("payment is Cash")
+                Statuses.FAILURE
+                return Check(cart.clientId, Statuses.FAILURE, ZERO_TOTAL_COSTS_ON_FAILED_BUY)
+            }
+
+            is Payment.CreditCard -> {
+                if (payment.expiryDate < LocalDate.now()) {
+                    // fail("expiry date is in the past")
+                    // update status to fail
+                    Statuses.FAILURE
+                } else if (payment.limit < totalPricesOfLegitProducts) {
+                    Statuses.FAILURE
+                } else if (payment.type !in listOf(CreditCardType.VISA, CreditCardType.MASTERCARD)) {
+                    Statuses.FAILURE
+                } else if (payment.number.length != VISA_LEGIT_NUM_LENGTH) {
+                    Statuses.FAILURE
+                } else { Statuses.SUCCESS }
+            }
+
+            is Payment.PayPal -> {
+                if (!payment.email.contains('@')) {
+                    Statuses.FAILURE
+                } else {
+                    Statuses.SUCCESS
+                }
+            }
+            else -> Statuses.SUCCESS
```

Comment:
Don't need else when you work with sealed class, you need to add package to types.kt

Reply:
I did import the sealed class to StoreService, did you mean anything else here - "you need to add package to types.kt"?
I don't understand why, but it seems that I tend to each class of the sealed class, and still the compiler ask for the "else" claiming it's not exhaustive. Tried even on the smalled example:
val status: Statuses = when(payment) {
is Payment.Cash -> Statuses.FAILURE
is Payment.CreditCard -> Statuses.FAILURE
is Payment.PayPal -> Statuses.FAILURE
} and it still didn't work. What do I miss here?

Reply:
you need to add package to your types.kt.